share/chruby/chruby.sh

```diff
@@ -1,10 +1,14 @@
 CHRUBY_VERSION="0.3.9"
 RUBIES=()
 
+saveglob="$(shopt -p nullglob dotglob)"
+shopt -s nullglob
+shopt -u dotglob
 for dir in "$PREFIX/opt/rubies" "$HOME/.rubies"; do
-	[[ -d "$dir" && -n "$(ls -A "$dir")" ]] && RUBIES+=("$dir"/*)
```

Comment:
Another thing to consider is the sort order. In zsh, you can get a proper numeric sort fairly easily, so `2.10` is greater than `2.9`.

Here is a zsh one-liner.
```sh
echo "${(@n)RUBIES}"
```
Or to print each Ruby in order:
```sh
printf "%s\n" "${(@n)RUBIES:t}"
```